### ISSUE TITLE
Why is the vector of "sentence_embedding" and the vector of the first token of "token_embeddings" in a model in cls mode different?

### ISSUE THREAD

**Issue by xalstnx** [UKPLab/sentence-transformers#2857]
hello. I am training and using a model.
The config.json in the pooling folder for this model is as follows.

{
  "word_embedding_dimension": 1024,
  "pooling_mode_cls_token": true,
  "pooling_mode_mean_tokens": false,
  "pooling_mode_max_tokens": false,
  "pooling_mode_mean_sqrt_len_tokens": false
}

And after loading the model, set output_value=None and output, the result is as follows.
{'input_ids': tensor([  101,  1045,  2031,  1037,  2047, 11416,  6024,  9850,   102],
       device='mps:0'), 'token_type_ids': tensor([0, 0, 0, 0, 0, 0, 0, 0, 0], device='mps:0'), 'attention_mask': tensor([1, 1, 1, 1, 1, 1, 1, 1, 1], device='mps:0'), 'token_embeddings': tensor([[ 0.0492,  0.0886, -0.2032,  ..., -0.2938,  1.3257,  0.3218],
        [-0.1441, -0.1813, -0.2510,  ..., -0.3058,  1.0303,  0.1465],
        [-0.0634, -0.1073, -0.0395,  ..., -0.3061,  1.0095,  0.1660],
        ...,
        [-0.0625, -0.2677, -0.2328,  ..., -0.1397,  1.1621,  0.3229],
        [ 0.0025, -0.0934, -0.4244,  ..., -0.3394,  0.7774,  0.4541],
        [-0.3658, -0.4477, -0.5051,  ..., -0.0577,  0.8805,  0.1738]],
       device='mps:0'), 'sentence_embedding': tensor([ 0.0029,  0.0051, -0.0118,  ..., -0.0170,  0.0769,  0.0187],
       device='mps:0')}

Why is the first vector in token_embeddings different from the vector in sentence_embedding?
* By default, normalization is set to be performed.

**Comment by ir2718** [comment]
Hi,

I'm getting exactly the same vectors using this code: 
```
from sentence_transformers import SentenceTransformer
from sentence_transformers.models import Transformer, Pooling

t = Transformer("bert-base-cased")
p = Pooling(
    word_embedding_dimension=512,
    pooling_mode_cls_token=True,
    pooling_mode_mean_tokens=False,
    pooling_mode_max_tokens=False,
    pooling_mode_mean_sqrt_len_tokens=False
)
s = SentenceTransformer(modules=[t, p])

sent_emb = s.encode("This is some text", output_value="sentence_embedding")
token_emb = s.encode("This is some text", output_value="token_embeddings").cpu().numpy()

diff = ((sent_emb - token_emb[0])**2).sum()

print(diff) # prints 0.0
```

Could you share the code you're using so I can reproduce the error?

**Comment by xalstnx** [comment]
thank you for the reply.
I used the basic sentence-transformer usage.

`from sentence_transformers import SentenceTransformer

model_path = "local_model_path"
model = SentenceTransformer(model_path, trust_remote_code=True)

data = "sentence~~"
query_vector = model.encode(data)
`

**Comment by xalstnx** [comment]
> Hi,
> 
> I'm getting exactly the same vectors using this code:
> 
> ```
> from sentence_transformers import SentenceTransformer
> from sentence_transformers.models import Transformer, Pooling
> 
> t = Transformer("bert-base-cased")
> p = Pooling(
>     word_embedding_dimension=512,
>     pooling_mode_cls_token=True,
>     pooling_mode_mean_tokens=False,
>     pooling_mode_max_tokens=False,
>     pooling_mode_mean_sqrt_len_tokens=False
> )
> s = SentenceTransformer(modules=[t, p])
> 
> sent_emb = s.encode("This is some text", output_value="sentence_embedding")
> token_emb = s.encode("This is some text", output_value="token_embeddings").cpu().numpy()
> 
> diff = ((sent_emb - token_emb[0])**2).sum()
> 
> print(diff) # prints 0.0
> ```
> 
> Could you share the code you're using so I can reproduce the error?

thank you for the reply.
I used the basic sentence-transformer usage.

`from` sentence_transformers import SentenceTransformer

model_path = "local_model_path"
model = SentenceTransformer(model_path, trust_remote_code=True)

data = "sentence~~"
query_vector = model.encode(data)

**Comment by tomaarsen** [comment]
Hello!
I think this is caused by a `Normalize` module being included in the model after the Pooling. You can `print(model)` to inspect all used modules. This `Normalize` module only normalizes the sentence embeddings.

See this code as an example:
```python
import torch
from sentence_transformers import SentenceTransformer
from sentence_transformers.models import Normalize

model_path = "mixedbread-ai/mxbai-embed-large-v1"
model = SentenceTransformer(model_path, trust_remote_code=True)

data = "sentence~~"
query_vector = model.encode(data, output_value=None)
embeddings_equal = torch.equal(query_vector["token_embeddings"][0], query_vector["sentence_embedding"])
print("Without a Normalize module\nAre the embeddings equal:", embeddings_equal)
print(query_vector["token_embeddings"][0, :5], query_vector["sentence_embedding"][:5])

model.add_module("normalize", Normalize())
query_vector = model.encode(data, output_value=None)
embeddings_equal = torch.equal(query_vector["token_embeddings"][0], query_vector["sentence_embedding"])
print("\nWith a Normalize module\nAre the embeddings equal:", embeddings_equal)
print(query_vector["token_embeddings"][0, :5], query_vector["sentence_embedding"][:5])
```
which outputs:
```
Without a Normalize module
Are the embeddings equal: True
tensor([ 0.1072, -0.5501,  0.1006,  0.5550, -0.0499], device='cuda:0') tensor([ 0.1072, -0.5501,  0.1006,  0.5550, -0.0499], device='cuda:0')

With a Normalize module
Are the embeddings equal: False
tensor([ 0.1072, -0.5501,  0.1006,  0.5550, -0.0499], device='cuda:0') tensor([ 0.0063, -0.0321,  0.0059,  0.0324, -0.0029], device='cuda:0')
```

- Tom Aarsen
